hooks/useRailgunTx.tsx

```diff
@@ -44,40 +33,17 @@ const useRailgunTx = () => {
     recipient: string;
   }) => {
     setIsShielding(true);
-    const resp =
-      args.tokenAddress === ethAddress
-        ? await shieldBaseToken(args)
-        : await shieldToken(args);
-    setIsShielding(false);
-    return resp;
-  };
-
-  const getGasDetailsSerialized = async (
-    gasEstimateString: string
-  ): Promise<TransactionGasDetailsSerialized> => {
-    const { maxFeePerGas, maxPriorityFeePerGas, gasPrice } = await provider
-      .getFeeData()
-      .then((data) => {
-        if (network === NetworkName.Polygon) {
-          // TODO: polygon gas
-        }
-        return data;
-      });
-    // evmGasType depends on the chain. BNB uses type 0.
-    if (evmGasType === EVMGasType.Type0)
-      return {
-        evmGasType,
-        gasEstimateString,
-        gasPriceString: gasPrice!.toHexString(),
-      };
-    if (evmGasType === EVMGasType.Type2)
-      return {
-        evmGasType,
-        gasEstimateString,
-        maxFeePerGasString: maxFeePerGas!.toHexString(), // Current gas Max Fee
-        maxPriorityFeePerGasString: maxPriorityFeePerGas!.toHexString(), // Current gas Max Priority Fee
-      };
-    throw new Error("Unsupported gas type for chain");
+    try {
+      const resp =
+        args.tokenAddress === ethAddress
+          ? await shieldBaseToken(args)
+          : await shieldToken(args);
+      setIsShielding(false);
+      return resp;
+    } catch (e) {
+      console.error(e);
```

Comment:
```suggestion
      throw e;
```

in fact, we could remove the trycatch entirely, and let the error automatically bubble up to caller

Reply:
The main reason I added the try catch was the form was stuck in the loading state because isShielding wasn't set back to false on errors. One other thing, I can trigger an error by cancelling the transaction in my wallet, and that was the main reason I moved from throwing to logging. I am happy to add back the throw, but we probably want to handle a transaction cancellation more gracefully.